src/main/scripts/src/app/shared/perspectives/documents/post-it/attribute-tree/attribute-tree.component.ts

```diff
@@ -31,14 +32,47 @@ export class AttributeTreeComponent {
   public children: object;
 
   @Input()
-  public root: boolean;
+  public editable: boolean;
 
-  public isArray = element => Array.isArray(element);
+  @Output()
+  public change: EventEmitter<string> = new EventEmitter();
 
-  public isString = element => typeof element === 'string';
+  /**
+   * Workaround because pipes were extremely slow
+   */
+  public keys(object: object): string[] {
+    return Object.keys(object);
+  }
 
-  public isStringArray = (element: object[]) => element.every(this.isString);
+  public isArray(element: any): boolean {
+    return Array.isArray(element);
+  }
 
-  public isObject = element => !this.isArray(element) && !this.isString(element);
+  public isLeaf(element: any): boolean {
+    return this.isNumber(element) || this.isString(element);
+  }
 
+  public isString(element: any): boolean {
+    return typeof element === 'string';
+  }
+
+  public isNumber(element: any): boolean {
+    return typeof element === 'number';
```

Comment:
Aren't there any utils which already provide such methods? I can see there is `isNumeric` in rxjs but there might also be similar ones directly in TypeScript. If there are no such utils, extract these methods to a separate `TypeChecker` util class.